### PR TITLE
TrayPublisher: Batch utils functions

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Just chore PR extracting parsing methods into separate reusable file. These functions should be refined further to make them more robust.

## Testing notes:
Use with https://github.com/ynput/OpenPype-premium/pull/30

### CODE REVIEW

**Comment by jakubjezek001** [comment]
Task linked: [OP-4760 Background batch must be rewritten for Tray publisher](https://app.clickup.com/t/6658547/OP-4760)

File: openpype/hosts/traypublisher/batch_parsing.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                                fields=["name"]):
+        if asset_doc["name"].lower() in asset_name.lower():
+            return get_asset_by_name(project_name, asset_doc["name"])
+
+
+def get_asset_by_name_case_not_sensitive(project_name, asset_name,
+                                         all_selected_asset_ids=None,
+                                         log=None):
+    """Handle more cases in file names"""
+    asset_name = re.compile(asset_name, re.IGNORECASE)
```
Comment (iLLiCiTiT) [inline openpype/hosts/traypublisher/batch_parsing.py:83]: Note: This won't work in AYON.

File: openpype/hosts/traypublisher/batch_parsing.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    return matching_asset_doc, version
+
+
+def parse_with_version(project_name, asset_name, version_regex,
+                       all_selected_asset_ids=None, log=None):
+    """Try to parse asset name from a file name containing version too
+
+    Eg. 'chair_v001.mov' >> 'chair', 1
+    """
+    log = _handle_log(log)
```
Comment (iLLiCiTiT) [inline openpype/hosts/traypublisher/batch_parsing.py:52]: I don't think one additional line deserve new function.
```suggestion
    if not log:
        log = Logger.get_logger(__name__)
```

File: openpype/hosts/traypublisher/batch_parsing.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        matching_asset_doc, version = (
+            parse_with_version(project_name, asset_name, version_regex,
+                               all_selected_asset_ids))
+
+    if matching_asset_doc is None:
+        matching_asset_doc = parse_containing(project_name, asset_name,
+                                              all_selected_asset_ids)
+
+    if matching_asset_doc is None:
+        raise CreatorError(
```
Comment (iLLiCiTiT) [inline openpype/hosts/traypublisher/batch_parsing.py:40]: I would maybe not raise `CreatorError` just return `None`. Creator still can create an instance with empty `asset`...

**Review by 64qam — APPROVED** [review]

**Pull request merged**